conan/tools/gnu/makedeps.py

```diff
@@ -0,0 +1,216 @@
+import os
+import re
+
+from conan.internal import check_duplicated_generator
+from conan.tools.files import save
+
+
+CONAN_MAKEFILE_FILENAME = "conandeps.mk"
+
+
+def _get_formatted_dirs(folders, prefix_path_, name):
+    """ Format the directories to be used in the makefile, adding the prefix path if needed
+    :param folders: list of directories
+    :param prefix_path_: prefix path
+    :param name: component name
+    :return: list of formatted directories
+    """
+    ret = []
+    for i, directory in enumerate(folders):
+        directory = os.path.normpath(directory).replace("\\", "/")
+        prefix = ""
+        if not os.path.isabs(directory):
+            prefix = f"$(CONAN_ROOT_{name})/"
+        elif directory.startswith(prefix_path_):
+            prefix = f"$(CONAN_ROOT_{name})/"
+            directory = os.path.relpath(directory, prefix_path_).replace("\\", "/")
+        ret.append(f"{prefix}{directory}")
+    return ret
+
+
+class MakeDeps(object):
+    """
+    Generates a Makefile with the variables needed to build a project with the specified.
+    """
+
+    _title = "# This Makefile has been generated by Conan. DO NOT EDIT!\n"
+
+    def __init__(self, conanfile):
+        """
+        :param conanfile: ``< ConanFile object >`` The current recipe object. Always use ``self``.
+        """
+        self._conanfile = conanfile
+
+    def _var_ref(self, var):
+        return f"$({var})"
+
+    def _var_assign_single(self, var, value):
+        return f"{var} = {value}"
+
+    def _var_assign_multi(self, var, value):
+        return " \\\n\t".join([f"{var} ="] + value)
+
+    def generate(self):
+        check_duplicated_generator(self, self._conanfile)
+
+        line_buffer = []
+        var_aggregates = {}
+
+        def _divert():
+            buf = list(line_buffer)
+            line_buffer.clear()
+            return buf
+
+        # Convert a name to Make-variable-friendly syntax
+        def _makeify(name):
+            return re.sub(r'[^0-9A-Z_]', '_', name.upper())
+
+        # Replace common prefixes in a path list with a variable reference
+        def _rootify(root, root_id, path_list):
+            root_len = len(root)
+            root_with_sep = root + os.path.sep
+            root_var_ref = self._var_ref(f"CONAN_ROOT_{root_id}")
+            return [root_var_ref + path[root_len:]  \
+                    if path.startswith(root_with_sep) else path for path in path_list]
+
+        # Output a variable assignment, appropriately formatted
+        def _output_var(name, item, root=None, root_id=None, prefix_var=None, blank_after=True):
+            if item is None:
+                return False
+            if type(item).__name__ in ('Component', 'SettingsItem', 'Version', 'str'):
```

Comment:
How are SettingsItem involved here?

I think this ``_output_var()`` function is quite complicated to understand what is doing.

Reply:
I once found a comment in a code, like: "Only God can understand this code"

I'll simplify by using Jinja templates. I need to design something better, because we have 3 levels here, global (CONAN_LIBS), dependencies (hellolib, zlib, ...) and their components. I could write a template for each and then unity them in a single file.